viewing_party/party.py

```diff
@@ -1,23 +1,206 @@
+import pytest
+from viewing_party.party import *
+from tests.test_constants import *
+
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie_dict = {}
+
+    if title and genre and rating:
+        movie_dict["title"] = title
+        movie_dict["genre"] = genre
+        movie_dict["rating"] = rating
+    else:
+        return None
+
+    return movie_dict
+
+
+def add_to_watched(user_data , movie):
+    user_data["watched"].append(movie)
+
+    return user_data
 
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+
+    return user_data
+
+
+def watch_movie(user_data, title):
+    for i in range(0,len(user_data["watchlist"])):
+        if title in user_data["watchlist"][i]["title"]:
+            removed_movie = user_data["watchlist"].pop(i)
+            user_data["watched"].append(removed_movie)
+
+    return user_data
+
+        
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    sum_rating = 0.0
+    for dict in user_data["watched"]:
+        sum_rating += dict["rating"]
+    
+    try:
+        avg_rating = sum_rating / len(user_data["watched"])
+    except ZeroDivisionError:
+        return 0.0
 
+    return avg_rating
+
+
+def get_most_watched_genre(user_data):
+    genre_dict = {}
+
+    if len(user_data["watched"]) == 0:
+        return None
+    
+    for dict in user_data["watched"]:
+        if not dict["genre"] in genre_dict:
+            genre_dict[dict["genre"]] = 1
+
+        elif dict["genre"] in genre_dict:
+            genre_dict[dict["genre"]] += 1
+    
+    most_watched = ["", 0]

```

Comment:
[recommendation] I'd make separate values (maybe `most_watched_genre` and `most_watched_genre_count`), rather than using a list like this. It can be hard to know what `most_watched[0]` and `most_watched[1]` represent.